### PR TITLE
feat: Cert failure handling

### LINKED ISSUE #523: Use name vs. username in notification emails
I was able to create a cert today on staging and got an email (nice work @SerpentBytes!):

Screenshot 2023-04-05 at 9 14 00 AM

Let's switch to use the user's name vs. username in the body of the email.

### PR DESCRIPTION
**Pull request by dadolhay**
This PR implements certificate flow error handling

Closes #523

### CODE REVIEW

File: app/queues/certificate/certificate-error-handler.server.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import type { CertificateJobData } from './certificateJobTypes.server';
+import type { QueueEventsListener } from 'bullmq';
+
+declare global {
+  var __cert_flow_error_handlers_init__: boolean;
+}
+
+/**
+ * When one step in teh queue fails, it fails all subsequent steps as well.
```
Comment (humphd) [inline app/queues/certificate/certificate-error-handler.server.ts:20]: s/teh/the

File: app/queues/certificate/certificate-error-handler.server.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+      await setIsReconciliationNeeded(true);
+
+      const certificateEntry = await certificateModel.getCertificateById(certificateId);
+
+      logger.debug('Sending failure notification email', { rootDomain, certificateId });
+      await addNotification({
+        emailAddress: certificateEntry.user.email,
+        subject: 'My.Custom.Domain certificate request failed',
+        message: `${certificateEntry.username}, your certificate request with domain: ${certificateEntry.domain} has failed. Log in to My.Custom.Domain to try again.`,
```
Comment (humphd) [inline app/queues/certificate/certificate-error-handler.server.ts:81]: Instead of `username`, let's use the user's `name` if possible.

**Review by humphd — CHANGES_REQUESTED** [review]
This looks good.  I've noted a few small things, but the logic and code seem great to me.

File: app/queues/certificate/certificate-error-handler.server.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dadolhay) [reply app/queues/certificate/certificate-error-handler.server.ts]: Sure!

**Review by humphd — APPROVED** [review]

**Review by SerpentBytes — APPROVED** [review]

**Review by Eakam1007 — APPROVED** [review]

**Comment by humphd** [comment]
After the fact reviews are also useful!

**Pull request merged**